src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -678,23 +679,24 @@ trait TypeDiagnostics {
                 || sym.isGetter && (sym.accessed.isVal || (sym.owner.isTrait && sym.hasFlag(STABLE)))
                 || sym.isLazy
             ) s"val ${sym.name.decoded}"
-            else if (sym.isSetter) { cond = valAdvice ; s"var ${sym.name.getterName.decoded}" }
+            else if (sym.isSetter) {
+              cond = valAdvice; s"var ${sym.name.getterName.decoded}"
+            }
             else if (sym.isMethod) s"method ${sym.name.decoded}"
             else if (sym.isModule) s"object ${sym.name.decoded}"
             else "term"
-            )
+          )
           typer.context.warning(pos, s"$why $what in ${sym.owner} $cond")
         }
+        for (defn: DefTree <- unusedPrivates.unusedTerms if shouldWarnOn(defn.symbol)) {
+          termWarning(defn)
+        }
```

Comment:
wow that's so pretty! It could even be a one-liner, if you're into that sort of thing.